src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -232,9 +232,9 @@ private static ReplicationBatchItem CreateReplicationBatchItem(DocumentsOperatio
             var data = GetCounterValuesData(context, ref tvh.Reader);
             var docId = ExtractDocId(context, ref tvh.Reader);
 
-            return new ReplicationBatchItem
+            return new CounterReplicationItem
             {
-                Type = ReplicationItemType.CounterGroup,
+                Type = ReplicationBatchItem.ReplicationItemType.CounterGroup,
```

Comment:
Why do we need this now?

Reply:
We are reading the `Type` from the stream, so basically we need those for backward compatibility.